deployments/openshift.yml

```diff
@@ -3,94 +3,99 @@
-# 3. Create the secrets referenced in the 'env' section under your repository Settings.
-# 4. Edit the 'branches' in the 'on' section to trigger the workflow on a push to your branch.
-# 5. Commit and push your changes.
-
-# For a more sophisticated example, see https://github.com/redhat-actions/spring-petclinic/blob/main/.github/workflows/petclinic-sample.yaml
-# Also see our GitHub organization, https://github.com/redhat-actions/
-# ▶️ See a video of how to set up this workflow at https://www.youtube.com/watch?v=6hgBO-1pKho
+# - Create an OpenShift app from the image and expose it to the internet
+
+# ℹ️ Before you begin:
+# 1. Have access to an OpenShift cluster. Refer to https://www.openshift.com/try
+# 2. The project you wish to add this workflow to should have a Dockerfile.
+#   - If you don't have a Dockerfile at the repository root,
+#     or if it is named differently (eg. 'Containerfile'), see the buildah-build step.
+#   - The buildah-build step can be replaced with any other image build step if you do not wish to use a Dockerfile.
+
+# ℹ️ This workflow requires some configuration.
+# 1. ✏️ Modify the top-level 'env' fields marked with the pencil ✏️ to fit your project.
+# 2. Create the OPENSHIFT_SERVER and OPENSHIFT_TOKEN secrets.
+#   - Refer to https://docs.github.com/en/actions/reference/encrypted-secrets
+#   - and https://github.com/redhat-actions/oc-login#readme
+# 3. Commit and push the workflow file to your default branch to trigger a workflow run.
+
+# 📖 Read a blog post explaining this workflow at https://www.openshift.com/blog/deploying-to-openshift-using-github-actions
+# ▶️ Watch a video of setting up this workflow at https://www.youtube.com/watch?v=6hgBO-1pKho
+# 👋 Visit our GitHub organization at https://github.com/redhat-actions/ to see our actions and provide feedback.
 
 name: OpenShift
 
-# ⬇️  Modify the fields marked with ⬇️ to fit your project, and create any secrets that are referenced.
-# https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets
 env:
-  # ⬇️ EDIT with your registry and registry path.
-  REGISTRY: quay.io/<username>
-  # ⬇️ EDIT with your registry username.
-  REGISTRY_USER: <username>
-  REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+  # ✏️ EDIT to set a name for your OpenShift app, or a default one will be detected.
+  APP_NAME: ""
+
+  # ✏️ EDIT with the port your application should be accessible on.
+  # If the container image exposes exactly one port, this can be left blank.
+  APP_PORT: ""
 
-  # ⬇️ EDIT to log into your OpenShift cluster and set up the context.
+  # ✏️ EDIT your repository secrets to log into your OpenShift cluster and set up the context.
   # See https://github.com/redhat-actions/oc-login#readme for how to retrieve these values.
   OPENSHIFT_SERVER: ${{ secrets.OPENSHIFT_SERVER }}
   OPENSHIFT_TOKEN: ${{ secrets.OPENSHIFT_TOKEN }}
```

Comment:
https://github.com/redhat-actions/oc-login/wiki/Using-a-Service-Account-for-GitHub-Actions